pkg/daemon/daemon.go

```diff
@@ -1601,9 +1612,99 @@ func PersistNetworkInterfaces(osRoot string) error {
 		return fmt.Errorf("failed to run nmstatectl: %w", err)
 	}
 
+	// If the node is using networking in the initrd (e.g. for Tang pinning), then
+	// it's possible that the network configuration there also relies on specific
+	// network names. To be safe, we'll add kargs to have the names also apply in
+	// the initrd, and then possibly remove them afterwards as part of cleanup.
+
+	if usesInitrdNetworking, err := HostUsesNetworkingInInitrd(); err != nil {
+		return err
+	} else if !usesInitrdNetworking {
+		glog.Info("Not modifying ifname= kargs; host doesn't use initrd networking")
+		return nil // nothing more to do!
+	}
+
+	var args []string
+	if pinning {
+		if ifnames, err = getIfnamesFromLinkFiles(osRoot); err != nil {
+			return err
+		}
+
+		// we just persisted NIC names; we'll need to add kargs
+		for ifname, mac := range ifnames {
+			args = append(args, "--append", "ifname="+ifname+":"+mac)
+		}
+	} else {
+		// We just cleaned up NIC names. For the links that remain, we want to keep
+		// their corresponding kargs. Note that `--delete` will only delete one instance
+		// of the kargs. So this is safe even if the user previously added those same
+		// kargs.
+		remainingIfnames, err := getIfnamesFromLinkFiles(osRoot)
+		if err != nil {
+			return err
+		}
+		for ifname, mac := range ifnames {
+			if _, ok := remainingIfnames[ifname]; !ok {
```

Comment:
maybe `s/ok/exists`

i.e. the test we are doing here is if a ifname that previously existed no longer exists right?

Reply:
`ok` is more idiomatic for map lookups but agree it helps here, so changed!